pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -709,6 +709,13 @@ func buildSidecarOutboundTCPListenerOptsForPortOrUDS(listenerMapKey *string,
 	if len(listenerOpts.bind) == 0 {
 		svcListenAddress := listenerOpts.service.GetAddressForProxy(listenerOpts.proxy)
 		svcExtraListenAddress := listenerOpts.service.GetExtraAddressesForProxy(listenerOpts.proxy)
+		// Override the svcListenAddress, using the proxy ipFamily, for cases where the ipFamily cannot be detected easily.
+		// For example: due to the possibility of using hostnames instead of ips in ServiceEntry,
+		// it is hard to detect ipFamily for such services.
+		if listenerOpts.service.Attributes.ServiceRegistry == provider.External && listenerOpts.proxy.IsIPv6() &&
```

Comment:
Not sure we need `listenerOpts.service.Attributes.ServiceRegistry == provider.External`, I think there are k8s cases that can fall under this

Reply:
Hi @howardjohn,  there are only 3 enumerates for the provider: `provider.Mock`, `provider.Kubernetes` and `provider.External`.  I have checked that `provider.External` should presents the services converted from service entry. So there should be no overlay for k8s cases.